File: code/mathematical-algorithms/Check_is_square/check_is_square.php
<?php
// Part of Cosmos by OpenGenus Foundation 

// function to check if a number is
function check($numbers)
{
  foreach ($numbers as $number)  // traversing through each number
  {
    $chk = (string)sqrt($number) ;  // sqrt returns float which is converted to string by explixit type casting
    if(strpos($chk,".")) // checking if there is decimal or not
    echo "$number is not a perfect square<br>";
    else
    echo "$number is a perfect square<br>";
  }
}

// array of numbers to be performed
$numbers = array(1,2,3,4,5,6,7,8,9,10);
check($numbers); // calling the function
 ?>

File: code/mathematical-algorithms/Check_is_square/check_is_square.cs
// Part of Cosmos by OpenGenus Foundation 
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace finding_is_square
{
    /// <summary>
    /// Class to check if number is a perfect square or not
    /// </summary>
    class is_square
    {
        /// <summary>
        /// an array to store numbers
        /// </summary>
        int[] numbers;
        /// <summary>
        /// constructor 
        /// </summary>
        /// <param name="number">numbers to be processed</param>
        public is_square(int[] number)
        {
            numbers = number;
        }
        /// <summary>
        /// method to check if perfect square or not
        /// </summary>
        public void check_sq()
        {
            foreach (int number in numbers) // traversing the whole integer array
            {
                if (Math.Sqrt(number).ToString().Contains("."))  // converting double to string to chk if it contains . or not
                    Console.WriteLine("number {0} is not a perfect square", number);
                else
                    Console.WriteLine("number {0} is a perfect square", number);
            }
        }
    }
    class Program
    {
        static void Main(string[] args)
        {
            // making an array of numbers to check
            int[] numbers = { 56, 90, 100, 80, 16 };
            
            // creating an object of is_square class and initializing with test inputs
            is_square sq = new is_square(numbers);

            sq.check_sq(); // checking
             
            Console.ReadKey(); // pausing
        }
    }
}
